File: src/dxaml/xcp/tools/XCPTypesAutoGen/TypeProxy.t.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

static const XCHAR* m_ppszKnownShortNames[] =
{
    //#comment  Generation tool will insert the short name for each listed type
    //#comment  L"<typeshortname>",
    //#insert TYPESHORTNAMES
};

static XBYTE m_uKnownShortNamesLength[] =
{
    //#comment  Generation tool will insert the character length of each listed short name
    //#comment  <typeshortnamelength>,
    //#insert TYPESHORTNAMELENGTHS
};

static const XCHAR* m_ppszKnownLongNames[] =
{
    //#comment  Generation tool will insert the short name for each listed type
    //#comment  L"<namespace.typeshortname>",
    //#insert TYPELONGNAMES
};

static XBYTE m_uKnownLongNamesLength[] =
{
    //#comment  Generation tool will insert the character length of each listed long name
    //#comment  <typelongnamelength>,
    //#insert TYPELONGNAMELENGTHS
};

static XBYTE m_pbTypeFlags[] =
{
    //#comment  Generation tool will insert the combined flags for each listed type
    //#comment  <typeflags>,
    //#insert TYPEFLAGS
};

static XUINT32 m_puCoreTypeIds[] =
{
    //#comment  Generation tool will insert the combined flags for each listed type
    //#comment  <typeindexname>,
    //#insert TYPEINDEXNAMES
};

File: dxaml/xcp/tools/XCPTypesAutoGen/TypeProxy.t.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using System;
using System.Collections.Generic;

namespace MS.Internal
{
    internal partial class TypeProxy
    {
        private static Type[] __types = 
        {
            null, 
            //#comment  Generation tool will insert control types
            //#comment  typeof(<namespace.controlname>),
            //#insert CONTROL_TYPES
        };

        private static CreateObjectDelegate[] __create = 
        { 
            null, 
            //#comment  Generation tool will insert a creation delegate for each type
            //#comment  () => { return new <namespace.controlname>(); },
            //#insert CREATION_DELEGATES
        };

        private static UInt32[] __coreIds = 
        { 
            0, 
            //#comment  Generation tool will insert the core ID for each type
            //#comment  CoreTypes.<namespace>_<controlname>ID,
            //#insert CONTROL_IDS
        };

        private static Dictionary<Type, UInt32> _typeMap = new Dictionary<Type, UInt32>(64);

        private static void InitializeStaticData()
        {
            for (int index = 1; index < __types.Length; ++index)
            {
                _typeMap[__types[index]] = (UInt32)index;
            }
        }
    }
}
